src/smolagents/agents.py

```diff
@@ -645,6 +661,297 @@ def __call__(self, task: str, **kwargs):
             answer += "\n</summary_of_work>"
         return answer
 
+    def save(self, output_dir):
+        """
+        Saves the relevant code files for your agent so it can be pushed to the Hub. This will copy the code of your
+        agent in `output_dir` as well as autogenerate:
+
+        - a `tools`folder containing the logic for each of the tools under `tools/{tool_name}.py`.
+        - an `agent.json` file containing a dictionary representing your agent.
+        - a `prompt.yaml` file containing the prompt templates used by your agent.
+        - an `app.py` file providing an UI for your agent when it is exported to a Space with `agent.push_to_hub()`
+        - a `requirements.txt` containing the names of the module used by your tool (as detected when inspecting its
+          code)
+
+        Args:
+            output_dir (`str`): The folder in which you want to save your tool.
+        """
+        # Recursively saved managed agents
+        if self.managed_agents:
+            for agent_name, agent in self.managed_agents.items():
+                os.makedirs(f"{output_dir}/{agent_name}", exist_ok=True)
```

Comment:
Even though `makedirs` is supposed to handle windows-non-windows conversation of separators correctly it is advised to use `os.path.join`, i.e.:
```py
os.makedirs(os.path.join(output_dir, agent_name), exist_ok=True)
```

Reply:
Sure. I didn't know if that's fine by you. Will add.